File: frontend/src/lib/utils/phone.ts
const PERSIAN_DIGITS = '۰۱۲۳۴۵۶۷۸۹'
const ARABIC_DIGITS = '٠١٢٣٤٥٦٧٨٩'

export function normalizeMobile(value: string): string {
  let v = value.trim()
  for (let i = 0; i < PERSIAN_DIGITS.length; i++) {
    v = v.replaceAll(PERSIAN_DIGITS[i]!, String(i))
    v = v.replaceAll(ARABIC_DIGITS[i]!, String(i))
  }
  v = v.replace(/[\s\-()]/g, '')
  if (v.startsWith('+98')) v = `0${v.slice(3)}`
  else if (v.startsWith('98') && v.length === 12) v = `0${v.slice(2)}`
  return v
}

export function isValidIranianMobile(value: string): boolean {
  return /^09\d{9}$/.test(normalizeMobile(value))
}

File: backend/app/utils/phone.py
import re

PERSIAN_DIGITS = "۰۱۲۳۴۵۶۷۸۹"
ARABIC_DIGITS = "٠١٢٣٤٥٦٧٨٩"


def normalize_mobile(value: str) -> str:
    v = value.strip()
    for i, p in enumerate(PERSIAN_DIGITS):
        v = v.replace(p, str(i))
    for i, p in enumerate(ARABIC_DIGITS):
        v = v.replace(p, str(i))
    v = re.sub(r"[\s\-()]", "", v)
    if v.startswith("+98"):
        v = "0" + v[3:]
    elif v.startswith("98") and len(v) == 12:
        v = "0" + v[2:]
    return v


def is_valid_iranian_mobile(value: str) -> bool:
    return bool(re.match(r"^09\d{9}$", normalize_mobile(value)))
